airflow/providers/salesforce/operators/tableau_refresh_workbook.py

```diff
@@ -0,0 +1,74 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.salesforce.hooks.tableau import TableauHook
+from airflow.utils.decorators import apply_defaults
+
+
+class TableauRefreshWorkbookOperator(BaseOperator):
+    """
+    Refreshes a Tableau Workbook/Extract
+
+    :param workbook_name: The name of the workbook to refresh.
+    :type workbook_name: str
+    :param site_id: The id of the site where the workbook belongs to.
+    :type site_id: str
+    :param tableau_conn_id: The Tableau Connection id containing the credentials
+        to authenticate to the Tableau Server.
+    :type tableau_conn_id: str
+    """
+
+    @apply_defaults
+    def __init__(self,
+                 workbook_name,
+                 *args,
+                 site_id='',
+                 tableau_conn_id='tableau_default',
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+        self.workbook_name = workbook_name
+        self.site_id = site_id
```

Comment:
Do you think it would be possible to have Tableau connection defined in `airflow/models/connection.py`
By having that the Extra field could have default site hence it would be able to extract the site from the connection rather than pass it every time the hook is used.
The hook itself can overwrite the default site in the connection.

Reply:
Good point.